src/m365/teams/commands/chat/chat-get.ts

```diff
@@ -0,0 +1,200 @@
+      if (!participants || participants.length === 0 || participants.some(e => !validation.isValidUserPrincipalName(e))) {
+        return `${args.options.participants} contains one or more invalid email addresses.`;
+      }
+    }
+
+    return true;
+  }
+  
+  private async getChatDetailsById(id: string): Promise<Chat> {
+    const requestOptions: AxiosRequestConfig = {
+      url: `${this.resource}/v1.0/chats/${encodeURIComponent(id)}`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'      
+    };
+
+    const chat = await request.get<Chat>(requestOptions);
+    return chat;
+  }
+
+  private async getChatIdByParticipants(participantsString: string): Promise<string> {
+    const participants = participantsString.toLowerCase().replace(/\s/g, '').split(',').filter(e => e && e !== '');
+    const currentUserEmail = accessToken.getUserNameFromAccessToken(Auth.service.accessTokens[this.resource].accessToken).toLowerCase();
+    const existingChats = await this.findExistingChatsByParticipants([currentUserEmail, ...participants]);
+    
+    if (!existingChats || existingChats.length === 0) {
+      throw new Error('No chat conversation was found with these participants.');
+    }
+
+    if (existingChats.length === 1) {
+      return existingChats[0].id as string;
+    }
+
+    const disambiguationText = existingChats.map(c => {
+      return `- ${c.id}${c.topic && ' - '}${c.topic} - ${c.createdDateTime && new Date(c.createdDateTime).toLocaleString()}`;
+    }).join(os.EOL);
+
+    throw new Error(`Multiple chat conversations with this topic found. Please disambiguate:${os.EOL}${disambiguationText}`);
+    
+  }
+  
+  private async getChatIdByName(name: string): Promise<string> {
+    const existingChats = await this.findExistingGroupChatsByTopic(name);
+
+    if (!existingChats || existingChats.length === 0) {
+      throw new Error('No chat conversation was found with this name.');
+    }
+
+    if (existingChats.length === 1) {
+      return existingChats[0].id as string;
+    }
+
+    const disambiguationText = existingChats.map(c => {
+      const memberstring = (c.members as ConversationMember[]).map(m => (m as AadUserConversationMember).email).join(', ');
+      return `- ${c.id} - ${c.createdDateTime && new Date(c.createdDateTime).toLocaleString()} - ${memberstring}`;
+    }).join(os.EOL);
+
+    throw new Error(`Multiple chat conversations with this topic found. Please disambiguate:${os.EOL}${disambiguationText}`);
+  }
+  
+  private async findExistingChatsByParticipants(expectedMemberEmails: string[]): Promise<Chat[]> {
+    const chatType = expectedMemberEmails.length === 2 ? "oneOnOne" : "group";
+    const endpoint = `${this.resource}/v1.0/chats?$filter=chatType eq '${chatType}'&$expand=members&$select=id,topic,createdDateTime,members`;
+    const foundChats: Chat[] = [];
+
+    const chats = await this.getAllChats(endpoint, []);
```

Comment:
We can use the `getAllItems` method from the `odata` helper class here

Reply:
Ah, nice, it's been refactored. We should refactor the `chat message send` command as well to make use of this. Is that something that needs to be created as an issue? I'm seeing multiple comments here that that command will benefit from.

Reply:
Good catch! Yes, let's create a separate issue for each refactoring so that we don't need to pile them up and can let others help us.

Reply:
https://github.com/pnp/cli-microsoft365/issues/3107